compiler_opt/tools/extract_ir.py

```diff
@@ -242,9 +304,16 @@ def extract_artifacts(obj: TrainingIRExtractor) -> Optional[str]:
 def main(argv):
   if len(argv) > 1:
     raise app.UsageError('Too many command-line arguments.')
-  flags.mark_flags_as_required(['output_dir', 'input'])
+  flags.mark_flags_as_required(['output_dir'])
+
   objs = []
-  if FLAGS.input_type == 'json':
+  if FLAGS.input is None:
+    if FLAGS.thinlto_build == 'lld':
+      objs = load_for_lld_thinlto(FLAGS.obj_base_dir, FLAGS.output_dir)
+    else:
+      logging.error('Input flag not provided')
+      raise ValueError
```

Comment:
don't we logging.fatal tho?

Reply:
I think the idea is an exception will already print its contents (the message), and logging.fatal isn't universal in killing the program (i.e. in test environments) so may as well just consistently use exceptions to kill the program everywhere

Reply:
That's the problem, exceptions won't either - if the surrounding code catches. It doesn't today, but it's pretty easy to do it later forgetting the intention (i.e. the developer intent - crash, no debate - can be missed by future developer)

We earlier gave the guidance to use logging.fatal instead of asserts, so let's be consistent here.

I am less worried about behavior in test environment for code that's equivalent to C/C++ CHECK macro elsewhere (e.g. absl), but I'd prefer having a solution that both fails everywhere, regardless of the environment. I'll look for a mechanism and then we can replace logging.fatal instances with that.

Reply:
Makes sense, I'll switch it over to logging.fatal here, but we do need a mechanism that kills consistently in a detectable way for test cases (in case the code isn't crash-inducing without logging.fatal). logging.fatal + raising was my previous solution, since we can test for it with `self.assertRaises()` in tests, and seems relatively straightforward as well, what do you think?